src/main/java/DukeParser.java

```diff
@@ -0,0 +1,116 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class DukeParser {
+    String inputString;
+    DukeCommand command;
+
+    public DukeParser(String inputString) {
+        this.inputString = inputString;
+    }
+
+    public DukeCommand getCommand() {
+        return this.command;
+    }
+
+    public String[] parse() {
+        parseCommand();
+        return parseCommandArgs();
+    }
+
+    private void parseCommand() {
+        // First split by space to get each individual word of the inputString
+        String[] splittedString = inputString.split(" ");
+
+        // The first(0th) word has to be a command.
+        String command = splittedString[0];
+
+        // Check whether the command is valid.
+        for (DukeCommand dukeCmd : DukeCommand.values()) {
+            if (dukeCmd.text.equals(command)) {
+                this.command = dukeCmd;
+                return;
+            }
+        }
+        // If command is invalid:
+        throw new Error("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
+
+    private String[] parseCommandArgs() {
+        List<String> commandArgs = new ArrayList<>();
+        switch (command) {
+            case LIST:
```

Comment:
Maybe consider adding a TODO comment to indicate that the code here is not ready?